src/main/scala/devices/debug/Debug.scala

```diff
@@ -894,14 +895,12 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
       DMSTATUSRdData.impebreak := (cfg.hasImplicitEbreak).B
     }
 
-    for (component <- 0 until nComponents ) {
-      when (~io.dmactive || ~dmAuthenticated) {
-        haveResetBitRegs(component) := false.B
-      }.elsewhen (hartIsInResetSync(component)) {
-        haveResetBitRegs(component) := true.B
-      }.elsewhen (io.innerCtrl.fire() && io.innerCtrl.bits.ackhavereset && hamaskWrSel(component)) {
-        haveResetBitRegs(component) := false.B
-      }
+    when(~io.dmactive || ~dmAuthenticated) {
+      haveResetBitRegs := 0.U
+    }.elsewhen (hartIsInResetSync.reduce(_ | _)) {
+      haveResetBitRegs := haveResetBitRegs | hartIsInResetSync.asUInt
+    }.elsewhen (io.innerCtrl.fire() && io.innerCtrl.bits.ackhavereset) {
+      haveResetBitRegs := haveResetBitRegs & (~(hamaskWrSel.asUInt))
```

Comment:
This logic is not the same as the original.  Suppose hart 23 is always in reset (maybe it is waiting for hart 0 to enable it).  The second clause will always take effect, preventing the third clause from handling ackhavereset for other harts.

Reply:
Hi,
Thank you for the inputs.
I wrongly assumed only 1 hart would be in 1 state at a time.  I'm looking into correcting this error.